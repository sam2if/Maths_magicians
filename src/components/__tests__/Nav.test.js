import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Navbar from '../Nav';

describe('Navbar', () => {
  it('renders Navbar correctly', () => {
    render(
      <Router>
        <Navbar />
      </Router>,
    );
    const homeLink = screen.getByRole('link', { name: 'Home' });
    const aboutLink = screen.getByRole('link', { name: 'Calculator' });
    const contactLink = screen.getByRole('link', { name: 'Quote' });

    expect(homeLink).toBeInTheDocument();
    expect(aboutLink).toBeInTheDocument();
    expect(contactLink).toBeInTheDocument();
  });
});
