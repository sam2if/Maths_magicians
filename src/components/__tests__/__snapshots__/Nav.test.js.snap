import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Navbar from '../Nav';

describe('Navbar', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <Router>
        <Navbar />
      </Router>,
    );
    expect(container).toMatchSnapshot();
  });
});
